### PR TITLE
Added script to generate operator 

### PR DESCRIPTION
**Pull request by aneeshkp**
This PR has scripts under hack/ folder. The script build-operator.sh to build operators.
There is ReadMe included inside hack folder.
The shell scripts help the end-user to build the operator and test their ansible templates and deploy the operators into a cluster.
We need to test it thoroughly because this is the script that will be used to generate operators via operator-sdk

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -8,18 +8,20 @@ ifeq ($(BUILD_VERBOSE),1)
...
 GOGET=$(GOCMD) get
 BINARY_NAME=helmExport
 BINARY_UNIX=$(BINARY_NAME)_unix
-HELM_EXAMPLE=./examples/helmcharts/nginx/
-ROLENAME=nginx
-WORKSPACE=workspace
+export GO111MODULE=on
+validate:
+	 @if [[ -z "${role}" && -z "${worspace}" && -z "${helm_chart}" ]]; then \
+			echo "Please set env valraibles, (source env.sh)"; \
```
Comment (ryandgoulding) [inline Makefile]: Typo for "variables".

File: hack/build-operator.sh [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  shift
+
+done
+
+#Useses Zetcd operator and build k8s operator
+#1.Export Zetcd Ansible
+#2.Remove Reference to tpl, include, template and sprig function
+# These are the cases not supported (include tuple for loops)
+#3.Build Ansible Operator-sdk
+#3 Copy Exported Ansible in to operator generated by Operator-SDK
```
Comment (ryandgoulding) [inline hack/build-operator.sh]: Should be 4?

File: hack/build-operator.sh [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+# Copy To  roles/{KNOWN ROLE}/defaults from roles/{KNOWN ROLE}/default
+# Copy to roles/{KNOWN ROLE}/templates from roles/{KNOWN ROLE}/templates
+# Copy to roles/{KNOWN ROLE}/tasks from roles/{KNOWN ROLE}/tasks
+
+#Might Need formating before copything thios over
```
Comment (ryandgoulding) [inline hack/build-operator.sh]: typo for "this".

File: hack/build-operator.sh [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+#Enable CLI argument passing
+#Parameter taken {operatorname} {role} --helmpath --workspace, --api-version --kind
+
+#Parameter taken operattor_name role --helmpath=../exmples/helmcharts/examplechart --workspace=./workspace, --api-version=aps.ztcd.com --kind=kind
```
Comment (ryandgoulding) [inline hack/build-operator.sh]: Tyope for "operator_name"

File: hack/build-operator.sh [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+if ! which operator-sdk 2>&1 >/dev/null; then
+  echo "Did not find operator-sdk, set INSTALL_OPERATOR_SDK=1 env and try againto install operator-sdk"
```
Comment (ryandgoulding) [inline hack/build-operator.sh]: "again to"

**Comment by ryandgoulding** [comment]
Running into issues on OSX:
```
ryangoulding@rgouldin-mac hack % ./build-operator.sh    
./build-operator.sh: line 5: usage: command not found
ryangoulding@rgouldin-mac hack % 
```

```
ryangoulding@rgouldin-mac hack % ./build-operator.sh -e
Operator for :
Role: nginx
Operator Name: nginx-operator
Helm location : ./examples/helmcharts/nginx
Output location : ./workspace
go build -o helmExport -v ./*.go
./helmExport export nginx --helm-chart=./examples/helmcharts/nginx --workspace=./workspace --generateFilters=true
**SNIPPED**        
Operator is genrated at /Users/ryangoulding/workspace/tmp/helm-ansible-template-exporter/hack/workspace/nginx-operator
/Users/ryangoulding/workspace/tmp/helm-ansible-template-exporter/hack/workspace/nginx-operator
find: -printf: unknown primary or operator
find: -printf: unknown primary or operator
find: -printf: unknown primary or operator
ryangoulding@rgouldin-mac hack % 
```

```
ryangoulding@rgouldin-mac hack % ./build-operator.sh -b 
Operator for :
Role: nginx
Operator Name: nginx-operator
Helm location : ./examples/helmcharts/nginx
Output location : ./workspace
-------------------------- --------------- 
           Build and publish operator        
-------------------------- --------------- 
sed: 1: "./deploy/operator.yaml": invalid command code .
ryangoulding@rgouldin-mac hack % 
```

**Comment by aneeshkp** [comment]
Fixed and Pushed

File: hack/build-operator.sh (hunk 3 above)
Comment (aneeshkp) [reply hack/build-operator.sh]: outdated

**Comment by aneeshkp** [comment]
> Running into issues on OSX:
> 
> ```
> ryangoulding@rgouldin-mac hack % ./build-operator.sh    
> ./build-operator.sh: line 5: usage: command not found
> ryangoulding@rgouldin-mac hack % 
> ```
> 
> ```
> ryangoulding@rgouldin-mac hack % ./build-operator.sh -e
> Operator for :
> Role: nginx
> Operator Name: nginx-operator
> Helm location : ./examples/helmcharts/nginx
> Output location : ./workspace
> go build -o helmExport -v ./*.go
> ./helmExport export nginx --helm-chart=./examples/helmcharts/nginx --workspace=./workspace --generateFilters=true
> **SNIPPED**        
> Operator is genrated at /Users/ryangoulding/workspace/tmp/helm-ansible-template-exporter/hack/workspace/nginx-operator
> /Users/ryangoulding/workspace/tmp/helm-ansible-template-exporter/hack/workspace/nginx-operator
> find: -printf: unknown primary or operator
> find: -printf: unknown primary or operator
> find: -printf: unknown primary or operator
> ryangoulding@rgouldin-mac hack % 
> ```
> 
> ```
> ryangoulding@rgouldin-mac hack % ./build-operator.sh -b 
> Operator for :
> Role: nginx
> Operator Name: nginx-operator
> Helm location : ./examples/helmcharts/nginx
> Output location : ./workspace
> -------------------------- --------------- 
>            Build and publish operator        
> -------------------------- --------------- 
> sed: 1: "./deploy/operator.yaml": invalid command code .
> ryangoulding@rgouldin-mac hack % 
> ```

The issue seems to be due macOS version of find not having -printf option. let me find an alternative

**Comment by aneeshkp** [comment]
fixed sed for mac, but left find cmd with pritnf . It will work in mac os if you install findutils.. as per stack overflow

**Comment by ryandgoulding** [comment]
Looks like the "hack/build-operator.sh -r" does not work in operator-sdk 0.17.0.  They changed the CLI.  I'll merge this and we can fix that later, as -r is just for testing anyway.

**Review by ryandgoulding — APPROVED** [review]

**Pull request merged**